internal/provider/event_collectors/install/splunk/audit_logs.go

```diff
@@ -0,0 +1,298 @@
+package installsplunk
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/p0-security/terraform-provider-p0/internal"
+	"github.com/p0-security/terraform-provider-p0/internal/common"
+)
+
+const (
+	SplunkIntegration  = "splunk"
+	AuditLogsComponent = "audit-log"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &AuditLogs{}
+var _ resource.ResourceWithImportState = &AuditLogs{}
+var _ resource.ResourceWithConfigure = &AuditLogs{}
+
+var stateAttribute = schema.StringAttribute{
+	Computed:            true,
+	MarkdownDescription: common.StateMarkdownDescription,
+}
+
+var HttpsPrefixRegex = regexp.MustCompile(`^https:`)
+var UuidRegex = regexp.MustCompile(`^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$`)
+var HecTokenClearTextKey = "hec_token_cleartext"
+
+func NewAuditLogs() resource.Resource {
+	return &AuditLogs{}
+}
+
+type AuditLogs struct {
+	installer *common.Install
+}
+
+type auditLogsModel struct {
+	State             types.String `tfsdk:"state"`
+	Token             types.String `tfsdk:"token"`
+	HecEndpoint       types.String `tfsdk:"hec_endpoint"`
+	HecTokenClearText types.String `tfsdk:"hec_token_cleartext"`
+	HecTokenHash      types.String `tfsdk:"hec_token_hash"`
+}
+
+type TokenReadWrite struct {
+	Hash      *string `json:"hash,omitempty"`
+	ClearText *string `json:"clearText,omitempty"`
+}
+
+type auditLogsJsonReadWrite struct {
+	State       *string         `json:"state,omitempty"`
+	HecEndpoint *string         `json:"endpoint,omitempty"`
+	HecToken    *TokenReadWrite `json:"token,omitempty"`
+}
+
+type auditLogsApiReadWrite struct {
+	Item auditLogsJsonReadWrite `json:"item"`
+}
+
+func (r *AuditLogs) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_audit_logs"
+}
+
+func (r *AuditLogs) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `An installation of the HTTP Event Collector`,
+		Attributes: map[string]schema.Attribute{
+			"state": stateAttribute,
+			"token": schema.StringAttribute{
+				Computed:            true,
```

Comment:
The provider automatically generates a random UUID for the token ID that does not need to be states in the `.tf` file.